### PR TITLE
0.3.x

### PR DESCRIPTION
**Pull request by derNarr**
List of changes / improvements:

* gzipped event files (solves #68)
* harmonize dict_ndl and ndl (solves #75)
* replace docopt with dependecy (solves #71)
* improve readme (solves #80)
* improve alpha attribute in dict_ndl (solves #73)

Still open:
* document Mac support ( #76)

### CODE REVIEW

File: doc/source/examples.rst [hunk 1]
```diff
@@ -2,15 +2,14 @@
 Examples
```
Comment (dekuenstle) [inline doc/source/examples.rst]: The general structure of the examples section is a bit confusing. 
We may want to differenciate better by moving to subpages between general example/explanation of discrimination learning, the module usage examples and minimal working examples (including data preprocessing and evaluation)

File: doc/source/examples.rst [hunk 2]
```diff
@@ -284,57 +330,63 @@ ds_s#_an_#h_ha_nd  hands
...
 Filter the events
-=================
+-----------------
 As we do not want to include the outcomes 'foot' and 'feet' in this example
 as well as their cues '#f', 'fo' 'oo', 'ot', 't#', 'fe', 'ee' 'et', we use the
 
 
 pyndl.preprocess module
------------------------
+^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (dekuenstle) [inline doc/source/examples.rst]: Rendered font looks like heading, but according to the text it should be just emphasised

File: doc/source/examples.rst [hunk 3]
```diff
@@ -347,37 +399,46 @@ make sure to go through its documentation.
...
 Learn the weights
-=================
+-----------------
 Computing the strength of associations for the data is now easy, using for
 example ``ndl.ndl`` from the
 
 
 pyndl.ndl module
-----------------
+^^^^^^^^^^^^^^^^
```
Comment (dekuenstle) [inline doc/source/examples.rst]: Rendered font looks like heading, but according to the text it should be just emphasised

File: doc/source/comparison_of_algorithms.rst [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+from observations for each trial :math:`t` of stimuli (cues) :math:`\vec{x}_t`
+and the reward (outcome) :math:`r_t`. According to
+[dayan_explaining_away_in_weight_space]_ one way to infer / estimate the
+distribution of the association vector :math:`\Pr(\vec{w}_t | r_1, \cdot, r_{t
+- 1}) \sim N(\hat{\vec{w}}, S_t)` is:
+
+.. math::
+
+   \hat{\vec{w}}_{t + 1} = \hat{\vec{w}}_t + \frac{ S_t \cdot
+   \vec{x}_t}{\vec{x}_t \cdot S_n \vec{x}_n + \tau^2} (r_t - \hat{\vec{w}}_t
```
Comment (dekuenstle) [inline doc/source/comparison_of_algorithms.rst]: Where does n come from?

File: doc/source/comparison_of_algorithms.rst [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+
+* equations above only for one reward / outcome not for a vector of rewards /
+  outcomes
+
+We can make the following identifications:
+
+=================  =============================================
+Kalman             Rescorla-Wagner
+=================  =============================================
+:math:`\vec{x}_t`  :math:`\vec{c}`
```
Comment (dekuenstle) [inline doc/source/comparison_of_algorithms.rst]: may add description of formula items too in table

File: README.rst [hunk 6]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
+===============
+These instructions will get you a copy of the project up and running on your
+local machine for development and testing purposes. See deployment for notes on
+how to deploy the project on a live system.
+
+
+Prerequisites
+-------------
+You need python 3.4 or newer and git installed on your machine. We recommend to
+install Minicoda (https://conda.io/miniconda.html) before installing ``pyndl``
```
Comment (dekuenstle) [inline README.rst]: Why miniconda?

File: README.rst [hunk 7]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
 
+If you want to inspect and change the source code as well as running tests and
+having the documentation at hand clone the repository and install the package
+in 'development' mode by running
+
+.. code:: bash
+
+    git clone https://github.com/quantling/pyndl.git
+    cd pyndl
     python setup.py develop
```
Comment (dekuenstle) [inline README.rst]: replace python with python3

File: README.rst [hunk 8]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
+
+Installing
+----------
+If you only want to use the package run you can install ``pyndl`` from pypi with:
 
-    python setup.py install [--user]
+.. code:: bash
 
-or development install with::
+    pip install --user pyndl
```
Comment (dekuenstle) [inline README.rst]: Replace pip with pip3

File: README.rst (hunk 7 above)
Comment (dekuenstle) [inline README.rst]: Usually you need sudo or --user here.

File: README.rst [hunk 9]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
     python setup.py develop
 
-in order to build a source package run::
 
-    python setup.py sdist
+Documentation and Examples
+--------------------------
+Documentation and examples can be found under
+http://pyndl.readthedocs.io/en/latest/ or in the ``doc/`` folder after cloning
+the repository.
```
Comment (dekuenstle) [inline README.rst]: The doc is not build on install. We may add the compile step to the install script, or explain here how to build the docs

Comment (dekuenstle) [inline README.rst]: We also should write, that `doc/build/html/index.html` is the entry point of the documentation

File: README.rst [hunk 10]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
+If you have installed ``pyndl`` by cloning the repository change directory to
+the repository and run (for fast tests):
 
+.. code:: bash
 
-Development
-===========
+    cd pyndl
+    py.test
+    py.test doc/souce/example.rst
```
Comment (dekuenstle) [inline README.rst]: or better `tox -e test`

File: README.rst [hunk 11]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
+Getting Started
+===============
+These instructions will get you a copy of the project up and running on your
+local machine for development and testing purposes. See deployment for notes on
+how to deploy the project on a live system.
+
+
+Prerequisites
+-------------
+You need python 3.4 or newer and git installed on your machine. We recommend to
```
Comment (dekuenstle) [inline README.rst]: For development you also need installed tox

File: README.rst [hunk 12]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
+see the `tags on this repository <https://github.com/quantling/pyndl/tags>`_.
 
 
-Testing
+Authors
 =======
-Check if the package does what it is supposed to do by running the test with
-``py.test``::
+See also the list of `contributors
+<https://github.com/quantling/pyndl/contributors>`_ who participated in this
```
Comment (dekuenstle) [inline README.rst]: TODO: file is missing

File: README.rst [hunk 13]
```diff
@@ -37,127 +37,117 @@ the Rescorla-Wagner learning rule to these corpora.
...
-environment and installs all dependencies. ::
+License
+=======
+This project is licensed under the MIT License - see the `LICENSE.txt
+<https://github.com/quantling/pyndl/blob/master/LICENSE.txt>`_ file for details
 
-    (pip install tox)
-    tox
 
+Acknowledgments
```
Comment (dekuenstle) [inline README.rst]: Should we also mention / cite ndl here?
ndl2 wasn't officially published, so there's no link?!

File: README.rst (hunk 10 above)
Comment (dekuenstle) [inline README.rst]: Usually in the README are informations for developers and endusers. 
CONTRIBUTING.rst is usually more about organisational stuff and rights  like PR management, acknowledgement ,...

File: tox.ini [hunk 14]
```diff
@@ -7,6 +7,7 @@ deps =
      test: pytest
 commands =
      test: py.test []
+     test: py.test doc/source/examples.rst
```
Comment (derNarr) [inline tox.ini]: It should check the source code within the examples.rst file. As tox is not working on my machine I tried to guess how to incorporate something like:

```
py.test doc/source/examples.rst
```

on my machine, which tests all the code chunks in `example.rst`.

File: README.rst (hunk 6 above)
Comment (derNarr) [inline README.rst]: Miniconda seems to be one of the easiest ways to get a good virtual environment for python within your userspace. What else should we suggest (going with the preinstalled python will not work for most distributions)?

File: README.rst (hunk 8 above)
Comment (derNarr) [inline README.rst]: In (Mini/Ana)conda it is ``pip``, under Arch it seems to be pip as well?! (nevertheless ``pip3`` should work in most cases anyway).

File: README.rst (hunk 10 above)
Comment (derNarr) [inline README.rst]: As tox is not working for me, it is difficult to test :)

File: README.rst (hunk 13 above)
Comment (derNarr) [inline README.rst]: It is planned to publish ndl2 (at some point in the future) to github. I only need time for it. Probably in August.

File: doc/source/examples.rst (hunk 2 above)
Comment (derNarr) [inline doc/source/examples.rst]: See #99

File: doc/source/examples.rst (hunk 3 above)
Comment (derNarr) [inline doc/source/examples.rst]: See #99

File: doc/source/comparison_of_algorithms.rst (hunk 4 above)
Comment (derNarr) [inline doc/source/comparison_of_algorithms.rst]: Fixed. Thanks!

File: doc/source/examples.rst [hunk 15]
```diff
@@ -2,15 +2,14 @@
 Examples
 ========
 
----------------
 Lexical example
----------------
+===============
 
 The lexical example illustrates the Rescorla-Wagner[@rescorlawagner1972]
```
Comment (derNarr) [inline doc/source/examples.rst]: See #99

File: doc/source/examples.rst (hunk 1 above)
Comment (derNarr) [inline doc/source/examples.rst]: See #99

File: doc/source/comparison_of_algorithms.rst (hunk 5 above)
Comment (derNarr) [inline doc/source/comparison_of_algorithms.rst]: See #99

File: README.rst (hunk 9 above)
Comment (dekuenstle) [inline README.rst]: See #99

**Pull request merged**